### PR TITLE
Update ConfigurationHolder.java

### LINKED ISSUE #50: Null Pointer Exception when no external configuration is supplied
**Describe the bug**
When none of .yaml,.json or .xml are supplied and we try to autowire a bean for AgentManager using AgentManager.defaultInstance() then a NullPointerException is thrown.
This happens because a container.getBean(..) is invoked in the fillAuxiliaryObjects() function call in ConfigurationHolder class and when none of the external configurations are present then getBean() returns null. This causes the following line of initialising agentsByClassName to throw a null pointer exceptin as the config is null.

**Expected behavior**
Even when no external configuration is supplied, a default instance of AgentManager should be wired with no agents and a global configuration. This is due to the fact that user may invoke scanPackages("..") programatically after a default instance is wired.

**Additional context**
Stack Trace:

`Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [net.obvj.agents.conf.ConfigurationHolder]: Constructor threw exception; nested exception is java.lang.NullPointerException: Cannot invoke "net.obvj.agents.conf.Configuration.getGlobalConfiguration()" because "config" is null
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:224) ~[spring-beans-5.3.16.jar:5.3.16]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:87) ~[spring-beans-5.3.16.jar:5.3.16]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1326) ~[spring-beans-5.3.16.jar:5.3.16]
	... 45 common frames omitted
Caused by: java.lang.NullPointerException: Cannot invoke "net.obvj.agents.conf.Configuration.getGlobalConfiguration()" because "config" is null
	at net.obvj.agents.conf.ConfigurationHolder.fillAuxiliaryObjects(ConfigurationHolder.java:96) ~[classes/:na]
	at net.obvj.agents.conf.ConfigurationHolder.<init>(ConfigurationHolder.java:61) ~[classes/:na]
	at net.obvj.agents.conf.ConfigurationHolder.<init>(ConfigurationHolder.java:55) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[na:na]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:211) ~[spring-beans-5.3.16.jar:5.3.16]
	... 47 common frames omitted`

### PR DESCRIPTION
**Pull request by academy-codex**
Fixes #50

### CODE REVIEW

**Review by oswaldobapvicjr — APPROVED** [review]

File: src/main/java/net/obvj/agents/conf/ConfigurationHolder.java [hunk 1]
```diff
@@ -86,6 +86,12 @@ private static TypeSafeConfigurationContainer<Configuration> defaultContainer()
     private void fillAuxiliaryObjects()
     {
         Configuration config = container.getBean();
+        // Set empty as defaults if none of .xml,.json or .yaml are supplied and caller will invoke scan packages programatically.
+        if (Objects.isNull(config)) {
```
Comment (oswaldobapvicjr) [inline src/main/java/net/obvj/agents/conf/ConfigurationHolder.java:90]: if (config == null)
{

**Review by oswaldobapvicjr — CHANGES_REQUESTED** [review]
Thanks for the finding.
I added a comment to check the nullity of `config` without using the `Objects` class because it's missing an import declaration.

File: src/main/java/net/obvj/agents/conf/ConfigurationHolder.java (hunk 1 above)
Comment (academy-codex) [reply src/main/java/net/obvj/agents/conf/ConfigurationHolder.java]: Done

File: src/main/java/net/obvj/agents/conf/ConfigurationHolder.java [hunk 2]
```diff
@@ -86,6 +86,12 @@ private static TypeSafeConfigurationContainer<Configuration> defaultContainer()
     private void fillAuxiliaryObjects()
     {
         Configuration config = container.getBean();
+        // Set empty as defaults if none of .xml,.json or .yaml are supplied and caller will invoke scan packages programatically.
+        if (config == null) {
+            agentsByClassName = new HashMap<>();
```
Comment (oswaldobapvicjr) [inline src/main/java/net/obvj/agents/conf/ConfigurationHolder.java:91]: The compilation is failing due to a missing import for `HashMap`.
Please try to use `Collections.emptyMap()` instead.


````
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project agents: Compilation failure
Error:  /home/runner/work/agents/agents/src/main/java/net/obvj/agents/conf/ConfigurationHolder.java:[91,37] cannot find symbol
Error:    symbol:   class HashMap
Error:    location: class net.obvj.agents.conf.ConfigurationHolder
Error:  -> [Help 1]
````

Comment (academy-codex) [reply src/main/java/net/obvj/agents/conf/ConfigurationHolder.java]: My bad, i edited in Web IDE. Took a pull, compiled and squashed commits.
Please check :)

**Review by oswaldobapvicjr — APPROVED** [review]

**Pull request merged**